solr/core/src/java/org/apache/solr/api/ContainerPluginsRegistry.java

```diff
@@ -372,9 +396,27 @@ public void init() throws Exception {
         holders.add(new ApiHolder((AnnotatedApi) api));
       }
     }
+
+    @SuppressWarnings("rawtypes")
+    private Class getConfigObj(Object o) {
+      Type[] interfaces = o.getClass().getGenericInterfaces();
```

Comment:
This is super fragile and fails if we have `Widget implements ConfigurablePlugin<WidgetInfo>` and then `NewerWidget extends Widget`

Reply:
Yeah, this can be improved

Reply:
fixed